src/Controls/src/SourceGen/RouteSyntaxReceiver.cs

```diff
@@ -0,0 +1,324 @@
+﻿using System.Collections.Generic;
```

Comment:
This file can be written in a much cleaner way with less nested if statements (which make it very hard to review and has a very high cyclomatic complexity). I went and did some example to clean the logic with switch and LINQ expressions and the code is a lot more readable: https://gist.github.com/mandel-macaque/98033a76ed54c9dd2bb559730b5caf6c

Reply:
Thanks for the feedback. Will look into it.

Reply:
@mandel-macaque Have pushed another commit that incorporates those feedback to improve the readability and maintainability.